### PR TITLE
Update arg parsing to clap v3

### PR DESCRIPTION
**Pull request by tranzystorek-io**
Updates argument parsing to the new version of clap, with its flagship `derive` feature that minimizes boilerplate code and automates type parsing.

Notable change: the verbosity flag seemed like a natural candidate for a `global`.

Also, by default clap 3 colors the help output, if that is unwanted it can be easily disabled.

### CODE REVIEW

**Comment by erikh** [comment]
I'll take a peek at this later tonight. Thank you very much for the patch!

**Comment by erikh** [comment]
Haven't forgotten about this, just waiting until monday :)

File: Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ edition = "2021"
 [dependencies]
 regex = ">= 0"
 anyhow = ">= 0"
-clap = "=2"
+clap = { version = "3.0", features = ["derive"] }
```
Comment (erikh) [inline Cargo.toml:19]: Let's make this `^3` until it breaks. I think that'll be better for catching bugfixes and security efforts.

File: src/main.rs [hunk 2]
```diff
@@ -29,206 +29,130 @@ mod integration_tests;
...
-            "No listening IPs for your interface; assign one in ZeroTier Central."
-        ));
+        return Ok(runtime.block_on(wait()));
     }
 
-    return Err(anyhow!("no network ID"));
+    return Err(anyhow!(
+        "No listening IPs for your interface; assign one in ZeroTier Central."
+    ));
+    // }
```
Comment (erikh) [inline src/main.rs:139]: Please remove this line

**Review by erikh — CHANGES_REQUESTED** [review]
Tentatively approving, aside from the outstanding comments. I still need to run tests (we don't have CI for this project yet).

For future patches, assuming you had a good experience and want to do more, please make sure you detail everything you changed. While appreciated (and arguably the right decision), the path/pathbuf stuff caught me by surprise and I'll need to take a deeper look at it with the tree merged.

**Comment by tranzystorek-io** [comment]
Thanks for taking your time! I'll gladly apply the suggestions.

> For future patches, assuming you had a good experience and want to do more, please make sure you detail everything you changed.

That's a good point, if you have any other questions I'd be happy to answer them 😄

**Comment by erikh** [comment]
Oh, no, I followed the patch, just not sure 100% how pathbuf in a few spots will affect the CLI usage of things, specifically around the options used less like authtoken. I just need to dink with it a little. No big deal, honest.

**Comment by erikh** [comment]
Patches look good. I want to play with this, so give me through tomorrow and then barring any surprises I'll merge it. Thanks again.

**Comment by erikh** [comment]
Ok, so the tests were failing to compile. I fixed that up and pushed it to your tree. Just figured it'd be simpler this way.

Thanks for the patch!

**Pull request merged**